app/controllers/admin/promo_registrations_controller.rb

```diff
@@ -0,0 +1,9 @@
+class Admin::PromoRegistrationsController < AdminController
+    def for_referral_code
```

Comment:
Curious, can we simply call the existing for_referral_code method that we've already defined instead of creating a new controller? 

I think that would prevent us from having to provide scope to all these different react components correct?

Reply:
I don't think this works since `for_referral_code` for publishers apply to that specific user, while `admin` scope works for all publishers.

Reply:
Seems straight forward enough to fix. The publisher's id is passed into the `for_referral_code` method already
The separation of the routes / additional code seems like it's harder to maintain than simply adding access parameters to the current solution

```ruby
if current_publisher.admin?
  publisher = Publisher.find(params[:id])
else 
  publisher = current_publisher
end
```